server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/GetAliasesResponse.java

```diff
@@ -76,4 +105,158 @@ public void writeTo(StreamOutput out) throws IOException {
             }
         }
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        GetAliasesResponse that = (GetAliasesResponse) o;
+        return Objects.equals(fromListAliasesToSet(aliases), fromListAliasesToSet(that.aliases));
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(fromListAliasesToSet(aliases));
+    }
+
+    private ImmutableOpenMap<String, Set<AliasMetaData>> fromListAliasesToSet(ImmutableOpenMap<String, List<AliasMetaData>> list) {
+        ImmutableOpenMap.Builder<String, Set<AliasMetaData>> builder = ImmutableOpenMap.builder();
+        list.forEach(e -> builder.put(e.key, new HashSet<>(e.value)));
+        return builder.build();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
As odd as this may sound, I think that we need to add `name` to the request (similar to what we do with `flatSettings` in some other requests, otherwise there is no way to filter aliases out when calling this API using the high-level REST client? That should also help making this `toXContent` method more sane, I think.

Reply:
I do not think it is necessary. The filtering can be done by specifying the aliases in the request `#aliases(String[])` , then they can be serialized as part of the url `/<indices>/_alias/<aliases>` and the rest layer will parse them as the `name`.

I introduce a `setName` in the request so you can have a look. Now there are two ways to 'filter' the returned aliases ( by specifying the aliases or the name )

Reply:
you are right, it is not necessary!